### ISSUE TITLE
Incorrect preprocessing behavior with line escaping

### ISSUE THREAD

**Issue by ZeWaka** [OpenDreamProject/OpenDream#1010]
Something is going wrong with how the preprocessor leaves stuff behind.

Test Case:
```
// setup
#define WOOF
/obj/h
/obj/i

// actual logic
/datum/foo
  var/ll = list(\
#ifdef WOOF
  /obj/h = 1,\
#endif
  /obj/i = 1)

/proc/main()
  var/datum/foo/x = new
  world.log << json_encode(x.ll)
```

BYOND output:
with define: `{"/obj/h":1,"/obj/i":1}`
without define: `{"/obj/i":1}`

OpenDream output:
with define: 
```
Error OD1100 at code.dm:28:7: There can only be whitespace before a preprocessor directive
Error OD0010 at code.dm:30:7: Unexpected #endif
```
without define: 
```
Error OD1100 at code.dm:27:7: There can only be whitespace before a preprocessor directive
Error OD0010 at code.dm:29:7: Unexpected #endif
Error OD0000 at code.dm:28:7: Expected ')'
Error OD0000 at code.dm:30:14: Expected end of object statement
Error OD0000 at code.dm:34:30: Invalid property "ll" on type /datum/foo
```

**Comment by amylizzle** [comment]
Yeah, this is fixed by #962

**Comment by wixoaGit** [comment]
#962 merged
